src/components/auth/store/AuthStore.js

```diff
@@ -51,38 +52,32 @@ const visibilitySlice = (set) => ({
 })
 
 const loginSlice = (set, get) => ({
-  login: () => {
-    const email = get().email
-    const password = get().password
-
-    ;(async () => {
-      // TODO: can we use apiClient here ?
-      try {
-        const response = await fetch('/api/v1/auth/login', {
-          method: 'POST',
-          headers: {
-            'Content-Type': 'application/json',
-          },
-          body: JSON.stringify({ email, password }),
-        })
-
-        if (response.ok) {
-          const data = await response.json()
-          const accessToken = data.jwt
-          const role = get().role
-          get().setUser({ email, role, accessToken })
-          get().hideAuthWidget()
-        }
-      } catch (error) {
-        console.error('Failed:', error)
-      }
-    })()
-  },
+  login: async () => {
+    const hasErrors = get().hasErrors;
+    if (hasErrors()) return;
+    const email = get().email;
+    const password = get().password;
+    try {
+      const response = await axios.post('/api/v1/auth/login', {
+        email,
+        password
+      });
```

Comment:
better replace this with apiClient from `src/components/services/` 
our api client applies couple more specific rules to each api call it makes:
- base path
- path headers

here, uri that you pass to axsios starts with "/api" - this is our "proxy" path 
configured by ourselves in `vite.config.js` and it might change any time
if changed we don't want to update it in every place all over the project, coz components should not know anything about it... 
only apiClient allowed to know it.